app/components/settings/billing-page/payment-history-section.hbs

```diff
@@ -0,0 +1,67 @@
+<Settings::FormSection @title="Payment history" @description="Details of your previous payments" data-test-payment-history-section>
+  {{#if this.isLoading}}
+    <div class="flex items-center justify-center py-8">
+      <img alt="Loading icon" src={{this.rippleSpinnerImage}} class="w-4 h-4" />
+      <span class="ml-3 text-gray-700 dark:text-gray-200">Loading payments...</span>
+    </div>
+  {{else if this.errorMessage}}
+    <div class="bg-red-50 dark:bg-red-900 border border-red-200 dark:border-red-400 rounded-md p-4">
+      <div class="flex gap-3">
+        <div class="flex-shrink-0">
+          {{svg-jar "x-circle" class="h-5 w-5 text-red-700 dark:text-red-300"}}
+        </div>
+        <p class="text-sm text-red-700 dark:text-red-300">{{this.errorMessage}}</p>
+      </div>
+    </div>
+  {{else if (gt this.charges.length 0)}}
+    <div>
+      <div class="grid grid-cols-3 gap-y-3">
+        {{! Header }}
+        <div class="border-b pb-2 text-gray-500 dark:text-gray-400 text-sm">Date</div>
```

Comment:
Border needs dark mode styling - here it's too bright

<img width="994" alt="Screenshot 2025-05-07 at 20 09 41" src="https://github.com/user-attachments/assets/e5f8e397-ee7c-4c6c-9ccd-83d9ec653789" />